controls/cf_serverd.cf

```diff
@@ -45,12 +45,24 @@ body server control
       cfruncommand => "$(sys.cf_agent) -I -D cfruncommand -f \"$(sys.update_policy_path)\"  &
                        $(sys.cf_agent) -I -D cfruncommand";
 
-    !windows::
-      # last single quote in cfruncommand is left open, so that
-      # arguments (like -K and --remote-bundles) are properly appended.
-      cfruncommand => "$(def.cf_runagent_shell) -c \'
-                           $(sys.cf_agent) -I -D cfruncommand -f $(sys.update_policy_path)  ;
-                           $(sys.cf_agent) -I -D cfruncommand";
+    !windows.!cfengine_3_7::
+
+      # in cfengine 3.10.0 and later last single quote in cfruncommand is left
+      # open, so that arguments (like -K and --remote-bundles) are properly
+      # appended.
+
+        cfruncommand => "$(def.cf_runagent_shell) -c \'
+                           $(sys.cf_agent) -I -f $(sys.update_policy_path)  ;
+                           $(sys.cf_agent) -I ";
```

Comment:
why did you remove `-Dcfruncommand`? If you want the class defined in 3.10, you have to append it manually. In older versions it used to be the case that cf-serverd was appending it before spawning cf-agent, when commanded via a remote cf-runagent.

Reply:
Here is what changed, copied from Core's changelog:

```
        - cf-serverd no longer appends "-I -Dcfruncommand" to
                     cfruncommand, this has to be done manually in masterfiles
                     body server control. (Redmine #7732)
```

Reply:
in other words, with CFEngine 3.10.x, if you want the class defined in cf-agent, when triggered via a remote cf-runagent, you have to manually append that in the policy.

If you remove it, it will result to the `cfruncommand` class being undefined when triggered by remote cf-runagent. Just this, nothing will blow up.

Reply:
Hmm... I worked on this a bit and mostly payed attention to getting the cfruncommand to work at all, not on the effect on the remote cf-agent's environment. Some more testing is in order?

Reply:
Why? Didn't it work before with 3.7.x ? I recall there were several fixes to cfruncommand parsing that were cherry-picked to 3.7.x.

Reply:
I'm not sure. Looks like Nick or I (or both) will have to dig back into this a bit.

Reply:
Good catch @jimis, I re-added the cfruncommand definition.